site/cypress/e2e/Cypress-Gradeable/late_submission_warning_messages.spec.js

```diff
@@ -0,0 +1,371 @@
+        // Create Gradeable
+        cy.get('#create-gradeable-btn').click();
+
+        //Date page, input 2 old dates for opening dates (Ta and students)
+        cy.get('#page_5_nav').click();
+        cy.get('#date_ta_view')
+            .clear()
+            .type('1992-06-15')
+            .type('{enter}');
+        cy.get('#save_status', {timeout:20000}).should('have.text', 'All Changes Saved');
+        cy.get('#date_submit')
+            .clear()
+            .type('2004-12-18')
+            .type('{enter}');
+        cy.get('#save_status', {timeout:20000}).should('have.text', 'All Changes Saved');
+    });
+
+    it('should show a warning message before late submission', () => {
+        //0 allowed late days and 0 remaining late days for student ==> Warning message
+        cy.login('instructor');
+        cy.visit(['sample', 'gradeable', gradeable, 'update?nav_tab=5']);
+        cy.get('#date_due')
+            .clear()
+            .type(getCurrentTime('few_seconds_future'))
+            .type('{enter}');
+        cy.get('#save_status', {timeout:20000}).should('have.text', 'All Changes Saved');
+        cy.logout();
+        SubmitAndCheckMessage('non_team', 'upload_file1', 'invalid_1_day_late');
+    });
+
+    it('should show a warning message before late submission', () => {
+        /*1 allowed late day and 0 remaining late day for student ==> Warning message
+        This is a basic case which is already included in part of the testing
+        below with extentions (3 days in the past test)
+        If testing runs for too long, you can remove this test bloc*/
+        cy.login('instructor');
+        cy.visit(['sample', 'gradeable', gradeable, 'update?nav_tab=5']);
+        cy.get('#late_days')
+            .clear()
+            .type(1,{force: true})
+            .type('{enter}');
+        cy.get('#save_status', {timeout:20000}).should('have.text', 'All Changes Saved');
+        cy.get('#date_due')
+            .clear()
+            .type(getCurrentTime())
+            .type('{enter}');
+        cy.get('#save_status', {timeout:20000}).should('have.text', 'All Changes Saved');
+        cy.logout();
+        SubmitAndCheckMessage('non_team', 'upload_file2', 'invalid_1_day_late');
+    });
+
+    it('should show a confirmation message before late submission', () => {
+        //1 allowed late day and 1 remaining late day for student ==> Confirmation message
+        cy.login('instructor');
+        giveLateDays(getCurrentTime(),'student');
+        cy.visit(['sample', 'gradeable', gradeable, 'update?nav_tab=5']);
+        cy.get('#date_due')
+            .clear()
+            .type(getCurrentTime())
+            .type('{enter}');
+        cy.get('#save_status', {timeout:20000}).should('have.text', 'All Changes Saved');
+        cy.logout();
+        SubmitAndCheckMessage('non_team', 'upload_file2', 'valid_usage', '1_day_late');
+    });
+
+    it('should show a warning message before late submission', () => {
```

Comment:
Could you make the 'it messages' a bit more specific, so we can more easily tell which part of the test failed? 
(Currently, if the 1 late day allowed, 1 late day remaining test failed, it would be hard to distinguish from the 0 days allowed, one remaining test in the CI)

Maybe 'Should show a warning message before late submission with 1 allowed 1 remaining' or something like that?